core_data_modules/cleaners/somali/demographic_patterns.py

```diff
@@ -1,4 +1,22 @@
+    }
+
+    yes_no = {
+        Codes.yes: r"\b\W*haa+h*\W*\b|\b\W*ye+s\W*\b|\b\W*haayes\W*\b|\b\W*xa+\W*\b",
+        Codes.no: r"\b\W*ma?ya\W*\b|\b\W*no\W*\b|\b\W*mayano\W*\b|\b\W*may[eo]*\W*\b|\b\W*m[ea]*y[ao]\W*\b|\bma\W*ya+g?\b|mayakalkal"
+    }
+
+    urban_rural = {
+        Codes.urban: r"m[as]+[qkg]a+l+[ao]+[ld]*a*|mglo|maga*lada|daynile|ma*ga+l[eo]?|m[ou][gq]a?dish[ou]|\bluuk|g[oa]r[ao]+w*e|qarawe|groe|burco|mudug|ma ?[gj]+a+l*o|mga+lo|yaqshi+d|cad*ale|bardhere|d[ae]yni*le|wadajir|[gk]a*lka?cyo|afgo*ye|daynile|ca*simada|harg[aei]+y?i?sr?a|m[as]+ga*l[eo]|city|capital|m[ou][gq]a?di?sh[aou]|[gk]a+l[gk]a?c?[iy]o|baidoa|kisi?ma+yo|jow?har|ha?rga?ys|bu+ca|baydhabo|baidoa|bay ?dha ?bo|beled ?weyn|b[ae]y\s?dha\s?b[ao]|afgoi|baardhe+re|b\Wdhe+re|baidoa|beydhuwa|baydhoba|baladweyne|b\Wd?\W?weyne|balad\s?we+yne|b\Wweyn|blwn|b[ae[l[ae]d\s?we+yne|bo+s+[ao]+s+o|bosso|o+sa+so|bu+ho+dle|bu+l+o\W*bur[dt]e|bu+rco|bu+hodle|cabudwa+q|ce+ga+g|ce+ri\W?ga+bo|\bcegbo|dhu\W?sa\smare+b|\beyl|ga+l*ka*c*yo|ka+lgac*yo|glkcy|galcad|gal[kc]aio|glkcyo*|ga[cl]kio|ga*lmu+du+g|ga+l[gka]cyo?|ga+l\s?kaciy?o|ka+lkacyo|ga+lkcayo|[ck]alkaca?yo|gur[ai]?ce+l|ha+ra?\s?g[ae]+[yi]+a?sa*|hage[iy]sa|\bhar\b|harga+i?[sy]+(a|eis|pa)?|harge+r?[iy]+si?a|har+gesi?a?|hargi+[ey]?sa|hargsa|hargyisa*|harkeysa|hasgeis|\bhay\sb\b|hayaey|hragayas|\bhrg\b|hrgaysa|hrgsa|hrgwysa|hrgys|hrysa|harge+a+|hargies|j[ao]+w*har*|kisi?m+ayo|ko+d\sbu+r|la+sqo+ray|lafo+le|\bluuq|m[ou][gkq]a*d[yi]sh*x*[uo]|town|balcad|bu\Wa+le|jama+me|megalo|b\Wdhera|balacad|bardhere|b[ae]ydhabo|bo?saso|ca+simad|celasha|mhaybe|shibis|dharkenley|gaga+lo|guri?ce+l|hiliwa|m[aw]ga+\s?lo|mega+l|muqdish[io]|salax\slafte+da|town|ca+simad|\b[jl]u+k|bosa+so|ka+ra+n|ko+dbu+r|afgo+ye|axmed\W*dhagax|b\W*w[ae]yn|ba+r\W*dhe+r|baidoa|balcad|bana+dir|bardhe+r|b[ae]ydhabo|bo+sa+s[ao]|bula burte|burtinle|c\W*wa+q|ca+s[iu]mad|cadada|caynabo|ce+lasha|ce+ri?ga+bo|celafwen|dagmada\W*warta|dayni+ne|dharkey?nl[ae]y|dhu\W*samare+b|gabilye|gacan\W*libax|gal\*kacyo|garbahare+y|gargaysa|guri?\W*[cl]e+l|harg[ae]ysa|hargso|hodon|hol\W*wadag|hudur|ibra+hi[mn]\W*ko+dbu|jilib|jowhar|karn|khanddala|kisma+yo|libaax gar|\bluuq|m\W*agalo|m\W*glo|ma+glada\W*bosaso|\bmaalo|madalada|madino|\bmagaa|magaglo|magala|m[ao]gal[od]|magolo|magool madaxda|malagada|maxamed haybe|mo?ga?la?da|miyi|m[ou]qdisho|mugadishu|qandala|saakow|tieglow|tuulada|wadajir|wanla weyn|war dhigley|warta nabada|x\W*jajab|xafada ak toobar",
```

Comment:
I don't find this especially easy to review. Let's add another issue for designing an operational checklist for reviewing regexes.

@as2388 @ritazagoni